pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -178,12 +181,40 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 	// If the usersignup is not banned and not deactivated then ensure the deactivated notification status is set to false.
 	// This is especially important for cases when a user is deactivated and then reactivated because the status is used to
 	// trigger sending of the notification. If a user is reactivated a notification should be sent to the user again.
+
 	if !banned && !userSignup.Spec.Deactivated {
 		if err := r.updateStatus(logger, userSignup, r.setStatusDeactivationNotificationUserIsActive); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
+	// If the usersignup is not banned and not within the pre-deactivation period then ensure the deactivation notification
+	// status is set to false. This is especially important for cases when a user is deactivated and then reactivated
+	// because the status is used to trigger sending of the notification. If a user is reactivated a notification should
+	// be sent to the user again.
+	if !banned && !states.Deactivating(userSignup) {
```

Comment:
Shouldn't we also check that the user is not deactivated? Otherwise, we will reset the status also for deactivated users, which we don't really need, do we?

Reply:
Now I understand - the deactivating status will be always set even when the `UserSignup` is deactivated. But it's not really clear from the code. It looks like two different states.

Reply:
I think when we move `Spec.Deactivated` to `Spec.State = [...]` we will need to auto delete `deactivating` from the state when setting it to `deactivated`.

Reply:
@alexeykazakov will include this in the state manager when I work on [CRT-1006](https://issues.redhat.com/browse/CRT-1006)